src/main/dolphinManager.ts

```diff
@@ -0,0 +1,216 @@
+          };
+
+          dolphin.on("close", async () => {
+            await fs.unlink(commFilePath);
+            this.dolphinInstances.playback = null;
+          });
+        }
+
+        if (this.dolphinInstances.playback.commFilePath) {
+          await fs.writeFile(this.dolphinInstances.playback.commFilePath, JSON.stringify(replayComm));
+        }
+
+        break;
+      }
+
+      case "spectate": {
+        if (index < 0) {
+          throw Error("Must have a valid index");
+        }
+        let dolphinInstance: DolphinInstance | undefined = find(this.dolphinInstances.spectate, ["index", index]);
+        if (!dolphinInstance?.dolphin) {
+          const commFilePath = await generateCommunicationFilePath(dolphinUseType);
+          const dolphin = await this.startDolphin(DolphinType.PLAYBACK, commFilePath, isoParams);
+          dolphinInstance = {
+            dolphinUseType: dolphinUseType,
+            index: index,
+            commFilePath: commFilePath,
+            dolphin: dolphin,
+          };
+
+          // create an array of DolphinInstances if this is our first time launching a spectator client
+          if (this.dolphinInstances.spectate === null) {
+            this.dolphinInstances.spectate = [dolphinInstance];
+          } else {
+            this.dolphinInstances.spectate.push(dolphinInstance);
+          }
+
+          dolphin.on("close", async () => {
+            await fs.unlink(commFilePath);
+            if (this.dolphinInstances.spectate) {
+              remove(this.dolphinInstances.spectate, (instance) => instance.index === index);
+            }
+          });
+        }
+        if (dolphinInstance.commFilePath) {
+          await fs.writeFile(dolphinInstance.commFilePath, JSON.stringify(replayComm));
+        }
+
+        break;
+      }
+
+      case "netplay": {
+        if (!this.dolphinInstances.netplay) {
+          const dolphin = await this.startDolphin(DolphinType.NETPLAY, undefined, isoParams);
+          this.dolphinInstances.netplay = {
+            dolphinUseType: "netplay",
+            dolphin: dolphin,
+          };
+
+          dolphin.on("close", () => {
+            this.dolphinInstances.netplay = null;
+          });
+        }
+
+        break;
+      }
+
+      case "config": {
+        let configureType = "";
+        if (dolphinType === DolphinType.NETPLAY) {
+          configureType = "configNetplay";
+        } else if (dolphinType === DolphinType.PLAYBACK) {
+          configureType = "configPlayback";
+        } else {
+          throw Error("Must define a dolphin type for configuration");
+        }
+
+        const dolphin = await this.startDolphin(dolphinType);
+        this.dolphinInstances[configureType] = {
+          dolphinUseType: "config",
+          dolphin: dolphin,
+        };
+
+        dolphin.on("close", () => {
+          this.dolphinInstances[configureType] = null;
+        });
+
+        break;
+      }
+    }
+  }
+
+  private async startDolphin(
```

Comment:
I think this could not be attached to the class since it's not public and not instance-specific. e.g. just having in the file somewhere
```
function async startDolphin(...) {
....
}
```